src/clarity-angular/data/datagrid/datagrid-row.spec.ts

```diff
@@ -151,6 +151,82 @@ export default function(): void {
                    expect(context.clarityDirective.selected).toBe(false);
                }));
 
+            it("selects the model on click only when `rowSelectionMode` is enabled (Single selection)", function() {
+                selectionProvider.selectionType = SelectionType.Single;
+                context.testComponent.item = {id: 1};
+                context.detectChanges();
+                const cell = context.clarityElement.querySelector("clr-dg-cell");
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.currentSingle).toBeUndefined();
+
+                // Enabling the rowSelectionMode
+                selectionProvider.rowSelectionMode = true;
+                context.detectChanges();
+                cell.click();
+                context.detectChanges();
+                expect(selectionProvider.currentSingle).toEqual(context.testComponent.item);
+            });
+
+            it("selects the model on click only when `rowSelectionMode` is enabled (Multi selection)", function() {
+                selectionProvider.selectionType = SelectionType.Multi;
+                context.testComponent.item = {id: 1};
+                context.detectChanges();
+                const cell = context.clarityElement.querySelector("clr-dg-cell");
+                expect(selectionProvider.current).toEqual([]);
+                cell.click();
```

Comment:
Why not just click the row? It seems more straightforward.

Reply:
Agree. Fixed.